src/main/java/uk/gov/hmcts/darts/arm/rpo/impl/ArmRpoApiImpl.java

```diff
@@ -59,8 +63,37 @@ public void getRecordManagementMatter(String bearerToken, Integer executionId, U
     }
 
     @Override
-    public IndexesByMatterIdResponse getIndexesByMatterId(String bearerToken, Integer executionId, UserAccountEntity userAccount) {
-        throw new NotImplementedException("Method not implemented");
+    public void getIndexesByMatterId(String bearerToken, Integer executionId, String matterId, UserAccountEntity userAccount) {
+        var armRpoExecutionDetailEntity = armRpoService.getArmRpoExecutionDetailEntity(executionId);
+        armRpoService.updateArmRpoStateAndStatus(armRpoExecutionDetailEntity, ArmRpoHelper.getIndexesByMatterIdRpoState(),
+                                                 ArmRpoHelper.inProgressRpoStatus(), userAccount);
+
+        StringBuilder errorMessage = new StringBuilder("Failure during ARM RPO get indexes by matter ID: ");
+        IndexesByMatterIdResponse indexesByMatterIdResponse;
+        try {
+            indexesByMatterIdResponse = armRpoClient.getIndexesByMatterId(bearerToken, createIndexesByMatterIdRequest(matterId));
+        } catch (FeignException e) {
+            // this ensures the full error body containing the ARM error detail is logged rather than a truncated version
+            log.error(errorMessage.append("Unable to get ARM RPO response") + " {}", e.contentUTF8());
+            throw handleFailureAndCreateException(errorMessage.toString(), armRpoExecutionDetailEntity, userAccount);
+        }
+
+        if (isNull(indexesByMatterIdResponse)
+            || CollectionUtils.isEmpty(indexesByMatterIdResponse.getIndexes())
+            || isNull(indexesByMatterIdResponse.getIndexes().getFirst())) {
```

Comment:
Can we be sure ARM will always return a list with at most one item?

Reply:
Yep not worried about the null/empty check, but I'm wondering can we ever get, say, 2 items in this list? If that's possible, would it always be the first item we want to get? One for @hemantasharma1129 to clarify perhaps?

Even if we only ever expect one value here, it might be worth logging a warning if size > 1, just so we capture it.

Reply:
I agree with @chrisbellingham-hmcts. ARM has confirmed it will be alway 1 object in the collection but its worth adding a warning if its more than 2.